### PR TITLE
Add CIDIU S.p.A. (Italy) source

### PR DESCRIPTION
**Pull request by glaaze**
Source for [CIDIU S.p.A.](https://cidiu.it) waste collection services for Turin northwest side. The provider doesn't offer an ics file, so we need BS4 to parse an html table which exposes the next 15 days schedule.

Test cases include every town served by CIDIU S.p.A.:

```
vscode ➜ /workspaces/hacs_waste_collection_schedule (Add-CIDIU-S.p.A-Italy) $  /usr/bin/env /usr/bin/python3 /home/vscode/.vscode-server/extensions/ms-python.debugpy-2024.10.0-linux-x64/bundled/libs/debugpy/adapter/../../debugpy/launcher 55287 -- /workspaces/hacs_waste_collection_schedule/custom_components/waste_collection_schedule/waste_collection_schedule/test/test_sources.py -s cidiu_it -l -i -t 
Testing source cidiu_it ...
  found 10 entries for Alpignano
    2024-08-12 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-14 : Indifferenziato [mdi:trash-can]
    2024-08-14 : Organico [mdi:food-apple]
    2024-08-17 : Organico [mdi:food-apple]
    2024-08-21 : Indifferenziato [mdi:trash-can]
    2024-08-21 : Organico [mdi:food-apple]
    2024-08-22 : Carta [mdi:newspaper-variant-multiple]
    2024-08-22 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-22 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-24 : Organico [mdi:food-apple]
  found 11 entries for Buttigliera Alta
    2024-08-12 : Indifferenziato [mdi:trash-can]
    2024-08-12 : Organico [mdi:food-apple]
    2024-08-14 : Carta [mdi:newspaper-variant-multiple]
    2024-08-15 : Organico [mdi:food-apple]
    2024-08-16 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-16 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-19 : Indifferenziato [mdi:trash-can]
    2024-08-19 : Organico [mdi:food-apple]
    2024-08-22 : Organico [mdi:food-apple]
    2024-08-22 : Vetro e lattine [mdi:bottle-wine]
    2024-08-23 : Sfalci abbonamento normale [mdi:leaf-cricle]
  found 10 entries for Coazze
    2024-08-13 : Organico [mdi:food-apple]
    2024-08-16 : Indifferenziato [mdi:trash-can]
    2024-08-16 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-17 : Organico [mdi:food-apple]
    2024-08-20 : Organico [mdi:food-apple]
    2024-08-22 : Carta [mdi:newspaper-variant-multiple]
    2024-08-22 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-22 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-23 : Indifferenziato [mdi:trash-can]
    2024-08-24 : Organico [mdi:food-apple]
  found 16 entries for Collegno
    2024-08-12 : Organico [mdi:food-apple]
    2024-08-12 : Carta [mdi:newspaper-variant-multiple]
    2024-08-13 : Indifferenziato [mdi:trash-can]
    2024-08-15 : Organico [mdi:food-apple]
    2024-08-16 : Indifferenziato [mdi:trash-can]
    2024-08-16 : Plastica [mdi:recycle]
    2024-08-16 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-19 : Organico [mdi:food-apple]
    2024-08-19 : Carta [mdi:newspaper-variant-multiple]
    2024-08-20 : Indifferenziato [mdi:trash-can]
    2024-08-22 : Organico [mdi:food-apple]
    2024-08-23 : Indifferenziato [mdi:trash-can]
    2024-08-23 : Plastica [mdi:recycle]
    2024-08-23 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-23 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-24 : Vetro e lattine [mdi:bottle-wine]
  found 10 entries for Druento
    2024-08-12 : Indifferenziato [mdi:trash-can]
    2024-08-12 : Organico [mdi:food-apple]
    2024-08-12 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-16 : Organico [mdi:food-apple]
    2024-08-19 : Indifferenziato [mdi:trash-can]
    2024-08-19 : Organico [mdi:food-apple]
    2024-08-20 : Carta [mdi:newspaper-variant-multiple]
    2024-08-22 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-22 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-23 : Organico [mdi:food-apple]
  found 10 entries for Giaveno
    2024-08-13 : Indifferenziato [mdi:trash-can]
    2024-08-13 : Organico [mdi:food-apple]
    2024-08-16 : Organico [mdi:food-apple]
    2024-08-16 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-20 : Indifferenziato [mdi:trash-can]
    2024-08-20 : Organico [mdi:food-apple]
    2024-08-22 : Carta [mdi:newspaper-variant-multiple]
    2024-08-22 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-22 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-23 : Organico [mdi:food-apple]
  found 11 entries for Grugliasco
    2024-08-13 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-14 : Indifferenziato [mdi:trash-can]
    2024-08-14 : Organico [mdi:food-apple]
    2024-08-17 : Organico [mdi:food-apple]
    2024-08-17 : Carta [mdi:newspaper-variant-multiple]
    2024-08-20 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-20 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-21 : Indifferenziato [mdi:trash-can]
    2024-08-21 : Organico [mdi:food-apple]
    2024-08-24 : Organico [mdi:food-apple]
    2024-08-24 : Carta [mdi:newspaper-variant-multiple]
  found 10 entries for Reano
    2024-08-14 : Indifferenziato [mdi:trash-can]
    2024-08-14 : Organico [mdi:food-apple]
    2024-08-14 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-17 : Organico [mdi:food-apple]
    2024-08-21 : Indifferenziato [mdi:trash-can]
    2024-08-21 : Organico [mdi:food-apple]
    2024-08-21 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-21 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-24 : Organico [mdi:food-apple]
    2024-08-24 : Carta [mdi:newspaper-variant-multiple]
  found 11 entries for Rivoli
    2024-08-14 : Indifferenziato [mdi:trash-can]
    2024-08-14 : Organico [mdi:food-apple]
    2024-08-14 : Carta [mdi:newspaper-variant-multiple]
    2024-08-14 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-17 : Organico [mdi:food-apple]
    2024-08-21 : Indifferenziato [mdi:trash-can]
    2024-08-21 : Organico [mdi:food-apple]
    2024-08-21 : Carta [mdi:newspaper-variant-multiple]
    2024-08-21 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-21 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-24 : Organico [mdi:food-apple]
  found 7 entries for Rosta
    2024-08-12 : Organico [mdi:food-apple]
    2024-08-16 : Organico [mdi:food-apple]
    2024-08-16 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-16 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-19 : Organico [mdi:food-apple]
    2024-08-23 : Organico [mdi:food-apple]
    2024-08-23 : Sfalci abbonamento normale [mdi:leaf-cricle]
  found 10 entries for Sangano
    2024-08-13 : Organico [mdi:food-apple]
    2024-08-14 : Indifferenziato [mdi:trash-can]
    2024-08-16 : Organico [mdi:food-apple]
    2024-08-16 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-20 : Organico [mdi:food-apple]
    2024-08-20 : Carta [mdi:newspaper-variant-multiple]
    2024-08-21 : Indifferenziato [mdi:trash-can]
    2024-08-22 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-22 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-23 : Organico [mdi:food-apple]
  found 10 entries for Trana
    2024-08-13 : Organico [mdi:food-apple]
    2024-08-13 : Carta [mdi:newspaper-variant-multiple]
    2024-08-14 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-15 : Indifferenziato [mdi:trash-can]
    2024-08-16 : Organico [mdi:food-apple]
    2024-08-20 : Organico [mdi:food-apple]
    2024-08-21 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-21 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-22 : Indifferenziato [mdi:trash-can]
    2024-08-23 : Organico [mdi:food-apple]
  found 11 entries for Venaria Reale
    2024-08-14 : Organico [mdi:food-apple]
    2024-08-14 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-14 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-15 : Indifferenziato [mdi:trash-can]
    2024-08-15 : Carta [mdi:newspaper-variant-multiple]
    2024-08-17 : Organico [mdi:food-apple]
    2024-08-21 : Organico [mdi:food-apple]
    2024-08-21 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-22 : Indifferenziato [mdi:trash-can]
    2024-08-22 : Carta [mdi:newspaper-variant-multiple]
    2024-08-24 : Organico [mdi:food-apple]
  found 13 entries for Villarbasse
    2024-08-12 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-12 : Sfalci abbonamento ridotto [mdi:leaf-circle-outline]
    2024-08-13 : Organico [mdi:food-apple]
    2024-08-14 : Indifferenziato [mdi:trash-can]
    2024-08-15 : Plastica [mdi:recycle]
    2024-08-15 : Vetro e lattine [mdi:bottle-wine]
    2024-08-16 : Organico [mdi:food-apple]
    2024-08-17 : Carta [mdi:newspaper-variant-multiple]
    2024-08-19 : Sfalci abbonamento normale [mdi:leaf-cricle]
    2024-08-20 : Organico [mdi:food-apple]
    2024-08-21 : Indifferenziato [mdi:trash-can]
    2024-08-22 : Plastica [mdi:recycle]
    2024-08-23 : Organico [mdi:food-apple]
```

### CODE REVIEW

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/cidiu_it.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+import requests
+from bs4 import BeautifulSoup
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "CIDIU S.p.A."
+DESCRIPTION = "Source for CIDIU waste collection services for the north-west Turin province"
+URL = "https://cidiu.it/"
+
+TEST_CASES = {
```
Comment (5ila5) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/cidiu_it.py:12]: Please reduce your amount of test cases a bit as running all source test takes unnecessary longer

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/cidiu_it.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Source:
+    def __init__(
+        self,
+        street=None,
+        street_number=None,
+        city=None,
```
[the remark this answers is not in the document]
Comment (glaaze) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/cidiu_it.py:111]: Yup, you're right. Those where leftovers from the source I've built my stuff around :)

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/cidiu_it.py (hunk 1 above)
Comment (glaaze) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/cidiu_it.py:12]: Reduced

**Comment by 5ila5** [comment]
Thanks for your contribution

I did some minor reformatting, allowed integer street_number and replaced logging with extra logger

**Pull request merged**